src/compiler/checker.ts

```diff
@@ -2061,7 +2061,16 @@ module ts {
                 }
                 // Handle export default expressions
                 if (declaration.kind === SyntaxKind.ExportAssignment) {

```

Comment:
I find it really bizarre that export assignments are handled in getTypeOfVariableOrParameterOrProperty, as an export assignment is neither a variable, nor a parameter, nor a property.